src/pages/national-insurance-number/app.ts

```diff
@@ -0,0 +1,66 @@
+import { APIGatewayProxyEvent, APIGatewayProxyResult } from 'aws-lambda';
+import { renderAsHtmlResponse } from './common/templating';
+import { getSession, updateSession } from './common/session';
+import { withErrorHandling } from './common/routing';
+import { Form, GatewayResult, parseForm } from './common/forms/forms';
+import { ErrorCollection } from './common/forms/errors';
+import { includes, NationalInsuranceNumber, NationalInsuranceNumberKnown } from './common/answer';
+
+const nationalInsuranceNumberKnownKey: keyof NationalInsuranceNumberKnown = 'national-insurance-number-known';
+const nationalInsuranceNumberKey: keyof NationalInsuranceNumber = 'national-insurance-number';
+const valid_options: ReadonlyArray<NationalInsuranceNumberKnown['national-insurance-number-known']> = ['yes', 'no'];
+
+export const lambdaHandler = withErrorHandling(async (event) => {
+    const method = event.httpMethod.toUpperCase();
+    if (method === 'GET') {
+        return get(event);
+    } else if (method === 'POST') {
+        return post(event);
+    } else {
+        return {
+            statusCode: 405,
+            body: JSON.stringify({
+                message: 'Method not allowed: ' + method,
+            }),
+        };
+    }
+});
+
+const get = async (event: APIGatewayProxyEvent): Promise<APIGatewayProxyResult> => {
+    const session = await getSession(event);
+    return renderAsHtmlResponse(event, 'template.njk', { session });
+};
+
+const post = (event: APIGatewayProxyEvent): GatewayResult =>
```

Comment:
This is nice, I assume we can use this pattern for other pages?

Reply:
That's the idea 🙂